test/integration/controller/core/queuedworkload_controller_test.go

```diff
@@ -0,0 +1,138 @@
+	"sigs.k8s.io/kueue/pkg/util/testing"
+	"sigs.k8s.io/kueue/test/integration/framework"
+)
+
+// +kubebuilder:docs-gen:collapse=Imports
+
+var _ = ginkgo.Describe("QueuedWorkload controller", func() {
+	var (
+		ns           *corev1.Namespace
+		queue        *kueue.Queue
+		clusterQueue *kueue.ClusterQueue
+	)
+
+	ginkgo.BeforeEach(func() {
+		ns = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "core-queuedworkload-",
+			},
+		}
+		gomega.Expect(k8sClient.Create(ctx, ns)).To(gomega.Succeed())
+	})
+
+	ginkgo.BeforeEach(func() {
+		clusterQueue = testing.MakeClusterQueue("clusterQueue_queuedworkload-controller").
+			Resource(testing.MakeResource(resourceGPU).
+				Flavor(testing.MakeFlavor(flavorModelA, "5").Ceiling("10").Obj()).
+				Flavor(testing.MakeFlavor(flavorModelB, "5").Ceiling("10").Obj()).Obj()).Obj()
+		queue = testing.MakeQueue("queue", ns.Name).ClusterQueue(clusterQueue.Name).Obj()
+		gomega.Expect(k8sClient.Create(ctx, queue)).To(gomega.Succeed())
+	})
+
+	ginkgo.AfterEach(func() {
+		gomega.Expect(framework.DeleteQueue(ctx, k8sClient, queue)).To(gomega.Succeed())
+		gomega.Expect(framework.DeleteClusterQueue(ctx, k8sClient, clusterQueue)).To(gomega.Succeed())
+	})
+
+	ginkgo.It("Should update status when workloads are created", func() {
+		cases := []struct {
+			workload  *kueue.QueuedWorkload
+			condition kueue.QueuedWorkloadCondition
+		}{
+			{
+				workload: testing.MakeQueuedWorkload("one", ns.Name).Request(corev1.ResourceCPU, "1").Obj(),
+				condition: kueue.QueuedWorkloadCondition{
+					Type:   kueue.QueuedWorkloadAdmitted,
+					Status: corev1.ConditionFalse,
+				},
+			},
+			{
+				workload: testing.MakeQueuedWorkload("two", ns.Name).Request(corev1.ResourceCPU, "1").Obj(),
+				condition: kueue.QueuedWorkloadCondition{
+					Type:   kueue.QueuedWorkloadAdmitted,
+					Status: corev1.ConditionFalse,
+				},
+			},
+			{
+				workload: testing.MakeQueuedWorkload("tree", ns.Name).Request(corev1.ResourceCPU, "1").Obj(),
+				condition: kueue.QueuedWorkloadCondition{
+					Type:   kueue.QueuedWorkloadAdmitted,
+					Status: corev1.ConditionFalse,
+				},
+			},
+			{
+				workload: testing.MakeQueuedWorkload("cuatro", ns.Name).Queue(queue.Name).
+					Request(corev1.ResourceCPU, "1").Obj(),
+				condition: kueue.QueuedWorkloadCondition{},
+			},
+		}
+
+		ginkgo.By("Creating workloads")
+		for _, tc := range cases {
+			gomega.Expect(k8sClient.Create(ctx, tc.workload)).To(gomega.Succeed())
+		}
+
+		for _, tc := range cases {
+			gomega.Eventually(func() bool {
+				var updatedQueueWorkload kueue.QueuedWorkload
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(tc.workload), &updatedQueueWorkload)).To(gomega.Succeed())
+				fmt.Println(updatedQueueWorkload.Status)
+				if tc.workload.Spec.Admission != nil {
+					return updatedQueueWorkload.Status.Conditions[0].Type == tc.condition.Type &&
+						updatedQueueWorkload.Status.Conditions[0].Status == tc.condition.Status
```

Comment:
The suggestion in https://github.com/kubernetes-sigs/kueue/pull/125#discussion_r831393177 was to also check the message so we test each case of the condition. There will be a different message for missing Q, missing CQ and "workload didn't fit"